server/base/src/main/java/org/apache/accumulo/server/AbstractServer.java

```diff
@@ -34,6 +35,7 @@
 import org.apache.accumulo.core.util.threads.Threads;
 import org.apache.accumulo.server.metrics.ProcessMetrics;
 import org.apache.accumulo.server.security.SecurityUtil;
+import org.apache.hadoop.util.ShutdownHookManager;
```

Comment:
What does using Hadoop's ShutdownHookManager buy us? It seems like it's further entrenching a dependency on Hadoop that doesn't give us much value. It's also not a stable API, so it can vary across Hadoop versions substantially (I don't know if it does, but it could, since Hadoop has marked it as "evolving").

If we're going to use something like this, I think we can achieve the same thing by just using the `Runtime.getRuntime().addShutdownHook(thread)` Java API directly.

Reply:
@keith-turner suggested using ShutdownHookManager, likely because it tries to bring some order to the chaos in that it runs the shutdown hooks in a priority based order. We are not using this in other places in our code and just calling `Runtime.addShutdownHook`. I need to evaluate those to see if there should be an implied order. If we want to send a signal to the VM to finish what it's doing then terminate normally (SIGTERM or SIGINT), then we would need to use the Hadoop ShutdownHookManager to ensure that our shutdown hooks fire and complete before shutting down the HDFS Filesytem.